ee/tabby-webserver/src/service/answer.rs

```diff
@@ -212,23 +227,91 @@ impl AnswerService {
         );
 
         let mut params = params.clone();
-        override_params
-            .as_ref()
-            .inspect(|x| x.override_params(&mut params));
+        if let Some(override_params) = override_params {
+            override_params.override_params(&mut params);
+        }
 
         match self.code.search_in_language(query, params).await {
-            Ok(docs) => docs.hits,
+            Ok(docs) => {
```

Comment:
please extract a function like: "merge_code_snippet", adding function comments to explain the current merging logic (if more than 1 snippet, content < 200 LoC)

Reply:
also add unit test